### DISCUSSION TITLE
How to save the image correctly?

### DISCUSSION THREAD

**Discussion by yyuuliang** [colour-science/colour-demosaicing#21] [Q&A] [answered]
hi
thanks for this lib. I have a question about saving img.
I can use the plot to plot a correct color img, however, I don't know how to save it correctly. The color in the saved img is quite different from the plotting. Will you please show a example of img saving?

**Comment by thomasmansencal** [comment, marked as answer]
Hi @yyuuliang,

So assuming you have an image:

```python
image = demosaicing_CFA_Bayer_Malvar2004(CFA)
```

It is float64 linear at this stage, so you could either save it as an EXR linear file:

```python
colour.write_image(image, 'my_image.exr')
```

or as a PNG file, but this would require non-linear encoding, e.g. sRGB:

```python
colour.write_image(colour.cctf_encoding(image), 'my_image.png')
```

**Reply by rakeshwar07** [reply]
Hi,

Thank you for this library. Most of the images you read and write are in .png format. Even the .cr2 format files were converted to .png. Is there any particular reason for that. I wonder wheater using .jpg files will have less quality image?

**Reply by thomasmansencal** [reply]
PNG can be loaded pretty much everywhere universally whilst being lossless as an image format. JPG would certainly reduce the image size but at the risk of introducing artefacts.

**Reply by vigneshgarrapally** [reply]
![my_image](https://user-images.githubusercontent.com/47147513/172386261-3a826026-75c6-4835-b647-3c7e85cdabad.png)
Hi I have  saved the light house image using `colour.write_image(colour.cctf_encoding(image), 'my_image.png')` which came out this way. could you help me where was the mistake?
Thanks

**Reply by thomasmansencal** [reply]
Hi @vigneshgarrapally,

It seems like the image you started with was already non-linearly encoded, please make sure that you are working with linear data, then for viewing (and storage in a 8-bit file) you will need to non-linearly encode it.

Cheers,

Thomas

**Reply by vigneshgarrapally** [reply]
Hi Thomas,
Thanks for the reply. Do you have any idea on how to convert non linear data to linear data. I am not aware of the differences actually. Currently, I am trying to demosaic the bayer data that I get from the machine vision cameras.

Regards
Vignesh

**Reply by thomasmansencal** [reply]
You would need to know how the image is encoded by the machine vision camera, they typically use a power function, ~gamma encoding. The manual should give you that information.

**Reply by vigneshgarrapally** [reply]
Thanks for the quick reply. I have gone through the docs and as you mentioned it is gamma encoding. How to convert it to send it to demosaicing algorithms. 

Regards
Vignesh

**Reply by thomasmansencal** [reply]
Using the `colour.gamma_function`:

```python
import colour
colour.gamma_function(0.18, 2.2)
```
